### PR TITLE
Add the library screen and the podcast details page to the TV app

### PR DESCRIPTION
**Pull request by chikoski**
This pull request adds the following two screen to the TV app of Jetcaster:

- Library screen: a screen to show the subscribed podcasts
- Podcast details screen: a screen to show all episodes of the selected podcast

Users are able to subscribe / unsubscribe the podcast from the podcast details screen.

### CODE REVIEW

File: Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/model/EpisodeList.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetcaster.tv.model
+
+import com.example.jetcaster.core.data.database.model.EpisodeToPodcast
+
+data class EpisodeList(val member: List<EpisodeToPodcast>) : List<EpisodeToPodcast> by member
```
Comment (arriolac) [inline Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/model/EpisodeList.kt:21]: Add `@Immutable` here?

Nice pattern using delegation here 👍

File: Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterApp.kt [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+}
+
+data class OverScanMarginSettings(
+    val default: OverScanMargin = OverScanMargin(),
+    val podcastDetails: OverScanMargin = OverScanMargin(top = 40.dp, bottom = 40.dp),
+    val drawer: OverScanMargin = OverScanMargin(start = 0.dp, end = 0.dp),
+    val catalog: OverScanMargin = OverScanMargin(start = 0.dp, end = 0.dp)
+)
+
+data class OverScanMargin(
```
Comment (arriolac) [inline Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterApp.kt:72]: Nit: Consider moving this under a package for themes. Perhaps a new file called `Spacing.kt` under `com.example.jetcaster.tv.ui.theme`?

File: Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterApp.kt [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+}
+
+internal data object JetcasterAppDefaults {
+    val overScanMargin = OverScanMarginSettings()
+    val gapSettings = GapSettings()
+    val cardWidth = CardWidth()
+    val padding = PaddingSettings()
+}
+
+data class OverScanMarginSettings(
```
Comment (arriolac) [inline Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterApp.kt:65]: Same here—might be better located in another theme-specific file.

File: Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterAppState.kt [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    data object Settings : Screen {
+        override val route: String = "settings"
+    }
+
+    data class Podcast(private val podcastUri: String) : Screen {
+        override val route = "$ROOT/$podcastUri"
+
+        companion object : Screen {
+            private const val ROOT = "podcast"
+            override val route = "$ROOT/{podcastUri}"
```
Comment (arriolac) [inline Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterAppState.kt:100]: How about declaring "podcastUri" as a variable? Something like:
```
val ARG_PODCAST_URI = "podcastUri"
```

File: Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterAppState.kt [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        }
+    }
+
+    data class Player(private val episodeUri: String) : Screen {
+        override val route = "$ROOT/$episodeUri"
+
+        companion object : Screen {
+            private const val ROOT = "player"
+            override val route = "$ROOT/{episodeUri}"
```
Comment (arriolac) [inline Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterAppState.kt:109]: Same here:

```
How about declaring "podcastUri" as a variable? Something like:

val ARG_PODCAST_URI = "episodeUri"
```

**Review by arriolac — APPROVED** [review]
Just some comments to improve reusability. Otherwise, LGTM

File: Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/model/EpisodeList.kt (hunk 1 above)
Comment (chikoski) [reply Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/model/EpisodeList.kt:22]: Good catch! Thanks!

File: Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterApp.kt (hunk 2 above)
Comment (chikoski) [reply Jetcaster/tv-app/src/main/java/com/example/jetcaster/tv/ui/JetcasterApp.kt]: That sounds good. Thanks for the suggestion.

**Pull request merged**